propertyestimator/layers/simulation.py

```diff
@@ -109,27 +164,161 @@ def __init__(self, physical_property, force_field_path, schema, options):
+
+                protocol = plugins.available_protocols[protocol_schema.type](protocol_schema.id)
+                protocol.schema = protocol_schema
+
+                template_value = template_values[index]
+
+                # Pass the template values to the target protocols.
+                for template_target in replicator.template_targets:
+
+                    if template_target.start_protocol != schema_to_replicate.id:
+                        continue
+
+                    indexed_target = ProtocolPath.from_string(
+                        template_target.full_path.replace('$index', str(index)))
+
+                    protocol.set_value(indexed_target, template_value)
+
+                self.schema.protocols[protocol_id] = protocol.schema
+
+        # Find any non-replicated protocols which take input from the replication templates,
+        # and redirect the path to point to the actual replicated protocol.
+        for protocol_id in self.schema.protocols:
+
+            protocol_schema = self.schema.protocols[protocol_id]
+
+            protocol = plugins.available_protocols[protocol_schema.type](protocol_schema.id)
+            protocol.schema = protocol_schema
+
+            # Look at each of the protocols inputs and see if its value is either a ProtocolPath,
+            # or a list of ProtocolPath's.
+            for required_input in protocol.required_inputs:
+
+                input_values = [value for value in protocol.get_value_references(required_input) if
+                                replicator.replicates_protocol_or_child(value)]
+
+                if len(input_values) == 0:
+                    continue
+
+                generated_path_list = {}
+
+                for input_value in input_values:
+
+                    # Replace the input value with a list of ProtocolPath's that point to
+                    # the newly generated protocols.
+                    path_list = [ProtocolPath.from_string(input_value.full_path.replace('$index', str(index)))
+                                 for index in range(len(template_values))]
+
+                    generated_path_list[input_value] = path_list
+
+                input_value = protocol.get_value(required_input)
+
+                if isinstance(input_value, ProtocolPath):
+
+                    protocol.set_value(required_input, generated_path_list[input_value])
+                    continue
+
+                new_list_value = []
+
+                for value in input_value:
+
+                    if not isinstance(value, ProtocolPath):
+
+                        new_list_value.append(value)
+                        continue
+
+                    new_list_value.extend(generated_path_list[value])
+
+                protocol.set_value(required_input, new_list_value)
+
+            # Update the schema of the modified protocol.
+            self.schema.protocols[protocol_id] = protocol.schema
 
     def _build_dependants_graph(self):
```

Comment:
Optional: "dependent" is the common spelling in U.S. English: https://grammarist.com/spelling/dependant-dependent/

Reply:
I think dependent is typically used as an adjective, and dependant is the noun - in this case because it's building a list of dependants (which are themselves dependent), I think dependant is correct? I'd be happy for this to go either way to be honest though, but this is probably beyond the scope of this PR as this is already established across the code base and should probably be proposed as a different PR?